### ISSUE TITLE
Make twisted/python/test/test_win32.py pass under Python 3

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#5818]
|[<img alt="vperic's avatar" src="https://avatars.githubusercontent.com/u/688281?s=50" width="50" height="50">](https://github.com/vperic)| @vperic reported|
|-|-|
|Trac ID|trac#5818|
|Type|task|
|Created|2012-07-28 11:22:14Z|
|Branch|https://github.com/twisted/twisted/tree/test_win32-5818|

As before, to actually make this actually run you need to import unittest from the library, not from trial, and then "python3.2 -m unittest twisted.python.test.test_win32" will run the tests.

The change is simple enough, to quote my commit message:

```
    t.p.win32: Do not import exceptions module
    
    The exceptions module has been removed in Python 3; exceptions are now
    stored in builtins. However, the code can be simplified by trying to
    assign a WindowsError and catching a NameError, which will work in all
    Python versions and allows the removal of a helper function.
```

Of course, this is all from running on a Linux machine, so there might be other errors also.

Attachments:

* [test_win32.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/7ce/7ceafebbccb556d1550c9bb38a93b4af8c486735/7c36ec0ae26966db3340714d3a9e9a15482bde19.patch) (837 bytes) - added by vperic on 2012-07-28 11:22:31Z - First version

<details><summary>Searchable metadata</summary>

```
trac-id__5818 5818
type__task task
reporter__vperic vperic
priority__normal normal
milestone__Python_3_x Python-3.x
branch__branches_test_win32_5818 branches/test_win32-5818
branch_author__vperic vperic
status__closed closed
resolution__fixed fixed
component__core core
keywords__py3k py3k
time__1343474534000000 1343474534000000
changetime__1343485445000000 1343485445000000
version__None None
owner__vperic vperic

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="pitrou's avatar" src="https://avatars.githubusercontent.com/u/1721820?s=50" width="50" height="50">](https://github.com/pitrou)<a name="note_1"></a>|@pitrou set owner to @vperic|
|-|-|

Looks fine to me.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun commented|
|-|-|

It would probably be good to actually run the tests, at least once, on Python 3 on Windows...  Otherwise, who knows if it actually passes (which the ticket summary explicitly calls for...)?

**Comment by twisted-trac** [comment]
|[<img alt="vperic's avatar" src="https://avatars.githubusercontent.com/u/688281?s=50" width="50" height="50">](https://github.com/vperic)<a name="note_3"></a>|@vperic commented|
|-|-|

(In [34984]) Create branch test_win32-5818

**Comment by twisted-trac** [comment]
|[<img alt="vperic's avatar" src="https://avatars.githubusercontent.com/u/688281?s=50" width="50" height="50">](https://github.com/vperic)<a name="note_4"></a>|@vperic commented|
|-|-|

Thanks, waiting on [buildbot](http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/test_win32-5818).

**Comment by twisted-trac** [comment]
|[<img alt="vperic's avatar" src="https://avatars.githubusercontent.com/u/688281?s=50" width="50" height="50">](https://github.com/vperic)<a name="note_5"></a>|@vperic commented|
|-|-|

Ok, tests pass (bar the usual failures on some of the Windows bots), so I'll go ahead and merge this as this module is a prerequisite for support a few more.

**Comment by twisted-trac** [comment]
|[<img alt="vperic's avatar" src="https://avatars.githubusercontent.com/u/688281?s=50" width="50" height="50">](https://github.com/vperic)<a name="note_6"></a>|@vperic set status to `closed`|
|-|-|

(In [34988]) Merge test_win32-5818: Make t.python.test/test_win32.py pass in Python 3

Author: vperic
Reviewer: antoine
Fixes: [#5818](https://github.com/twisted/twisted/issues/5818)

t.p.win32: Do not import exceptions module

The exceptions module has been removed in Python 3; exceptions are now
stored in builtins. However, the code can be simplified by trying to
assign a WindowsError and catching a NameError, which will work in all
Python versions and allows the removal of a helper function.
